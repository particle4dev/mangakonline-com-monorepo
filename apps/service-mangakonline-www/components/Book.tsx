import * as React from 'react';
import Link from 'next/link';
import CardActionArea from '@material-ui/core/CardActionArea';
import { withStyles, createStyles } from '@material-ui/core/styles';
import Card, { CardProps } from '@material-ui/core/Card';

export type BookProps = CardProps & {
  href?: string,
  as?: string
};

const Book = withStyles(() =>
  createStyles({
    root: {
      background: 'none'
    }
  }),
)(({children, href, as, ...other}: BookProps) => href ? <Link href={href} as={as}>
  <a style={{
    textDecoration: 'none'
  }}>
    <Card elevation={0} {...other}>
      <CardActionArea>{children}</CardActionArea>
    </Card>
  </a>
</Link> : <Card elevation={0} {...other}>
  <CardActionArea>{children}</CardActionArea>
</Card>);

if (process.env.NODE_ENV !== 'production') {
  Book.displayName = 'components__Book';
}

Book.defaultProps = {};

export default Book;
