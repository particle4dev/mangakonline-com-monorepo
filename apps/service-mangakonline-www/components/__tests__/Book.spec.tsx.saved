import * as React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Book from '../Book';

describe('Book', () => {
  it('should render successfully', () => {
    const { baseElement, getByText } = render(<Book>test</Book>);
    expect(getByText('test')).toBeInTheDocument();
    // Snapshot demo
    expect(baseElement).toMatchSnapshot();
  });
});
