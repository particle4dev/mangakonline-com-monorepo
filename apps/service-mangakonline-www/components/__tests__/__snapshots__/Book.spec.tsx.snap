// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book should render successfully 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root MuiCard-root Component-root-1 MuiPaper-elevation0 MuiPaper-rounded"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root"
        tabindex="0"
        type="button"
      >
        test
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</body>
`;
